GitFyle.Core.Api.Tests.Acceptance/Apis/Repositories/RepositoriesApiTests.Logic.cs

```diff
@@ -0,0 +1,119 @@
+using GitFyle.Core.Api.Tests.Acceptance.Models.Sources;
+
+namespace GitFyle.Core.Api.Tests.Acceptance.Apis.Repositories
+{
+    public partial class RepositoriesApiTests
+    {
+        [Fact]
+        public async Task ShouldPostRepositoryAsync()
+        {
+            // given
+            Source randomSource = await PostRandomSourceAsync();
+            Repository randomRepository = await PostRandomRepository(sourceId: randomSource.Id);
+            Repository inputRepository = randomRepository;
+            Repository expectedRepository = inputRepository.DeepClone();
+
+            // when
+            Repository actualRepository =
+                await this.gitFyleCoreApiBroker.GetRepositoryByIdAsync(inputRepository.Id);
+
+            // then
+            actualRepository.Should().BeEquivalentTo(expectedRepository);
+            await this.gitFyleCoreApiBroker.DeleteRepositoryByIdAsync(actualRepository.Id);
+            await this.gitFyleCoreApiBroker.DeleteSourceByIdAsync(randomSource.Id);
+        }
+
+        [Fact]
+        public async Task ShouldGetRepositoryByIdAsync()
+        {
+            // given
+            Source randomSource = await PostRandomSourceAsync();
+            Repository randomRepository = await PostRandomRepository(sourceId: randomSource.Id);
+            Repository inputRepository = randomRepository;
+            Repository expectedRepository = inputRepository.DeepClone();
+
+            // when
+            Repository actualRepository =
+                await this.gitFyleCoreApiBroker.GetRepositoryByIdAsync(inputRepository.Id);
+
+            // then
+            actualRepository.Should().BeEquivalentTo(expectedRepository);
+            await this.gitFyleCoreApiBroker.DeleteRepositoryByIdAsync(actualRepository.Id);
+            await this.gitFyleCoreApiBroker.DeleteSourceByIdAsync(randomSource.Id);
+        }
+
+        [Fact]
+        public async Task ShouldGetAllRepositoriesAsync()
+        {
+            // given
+            Source randomSource = await PostRandomSourceAsync();
+
+            List<Repository> inputRepositories =
+                await PostRandomRepositoriesAsync(sourceId: randomSource.Id);
+
+            IEnumerable<Repository> expectedRepositories = inputRepositories;
+
+            // when
+            IEnumerable<Repository> actualRepositories =
+                await this.gitFyleCoreApiBroker.GetAllRepositoriesAsync();
+
+            // then
+            foreach (Repository expectedRepository in expectedRepositories)
+            {
+                Repository actualRepository = 
+                    actualRepositories.Single(repository => repository.Id == expectedRepository.Id);
+
+                actualRepository.Should().BeEquivalentTo(expectedRepository);
+                await this.gitFyleCoreApiBroker.DeleteRepositoryByIdAsync(actualRepository.Id);
+            }
+            await this.gitFyleCoreApiBroker.DeleteSourceByIdAsync(randomSource.Id);
```

Comment:
New line after multiline code block

```suggestion
            }

            await this.gitFyleCoreApiBroker.DeleteSourceByIdAsync(randomSource.Id);
```

Reply:
Done